default/AI/CombatRatingsAI.py

```diff
@@ -0,0 +1,364 @@
+            self._fighter_stats = self.FighterStats(stats[3:])
+        else:
+            self._basic_stats = self.BasicStats()
+            self._fighter_stats = self.FighterStats()
+            self.__get_stats_from_ship()
+
+    def __str__(self):
+        return str(self.get_stats())
+
+    def __get_stats_from_ship(self):
+        """Read and store combat related stats from ship"""
+        universe = fo.getUniverse()
+        ship = universe.getShip(self.__ship_id)
+        if not ship:
+            return  # TODO: Add some estimate for stealthed ships
+
+        if self._consider_refuel:
+            structure = ship.currentMeterValue(fo.meterType.maxStructure)
+            shields = ship.currentMeterValue(fo.meterType.maxShield)
+        else:
+            structure = ship.currentMeterValue(fo.meterType.structure)
+            shields = ship.currentMeterValue(fo.meterType.shield)
+        attacks = {}
+        fighter_launchrate = 0
+        fighter_capacity = 0
+        fighter_damage = 0
+        design = ship.design
+        if design and ship.isArmed:
+            meter_choice = fo.meterType.maxCapacity if self._consider_refuel else fo.meterType.capacity
+            for partname in design.parts:
+                if not partname:
+                    continue
+                pc = get_part_type(partname).partClass
+                if pc == fo.shipPartClass.shortRange:
+                    damage = ship.currentPartMeterValue(meter_choice, partname)
+                    attacks[damage] = attacks.get(damage, 0) + 1
+                elif pc == fo.shipPartClass.fighterBay:
+                    fighter_launchrate += ship.currentPartMeterValue(fo.meterType.capacity, partname)
+                elif pc == fo.shipPartClass.fighterHangar:
+                    fighter_capacity += ship.currentPartMeterValue(meter_choice, partname)
+                    part_damage = ship.currentPartMeterValue(fo.meterType.secondaryStat, partname)
+                    if part_damage != fighter_damage and fighter_damage > 0:
+                        # the C++ code fails also in this regard, so FOCS content *should* not allow this.
+                        # TODO: Depending on future implementation, might actually need to handle this case.
+                        print "WARNING: Multiple hangar types present on one ship, estimatates expected to be wrong."
+                    fighter_damage = max(fighter_damage, part_damage)
+        self._basic_stats = self.BasicStats(structure=structure, shields=shields, attacks=attacks)
+        self._fighter_stats = self.FighterStats(fighter_capacity, fighter_launchrate, fighter_damage)
+
+    def get_basic_stats(self, hashable=False):
+        """Get non-fighter-related combat stats of the ship.
+
+        :param hashable: if true, returns tuple instead of attacks-dict
+        :return: attacks, structure, shields
+        """
+        return self._basic_stats.get_stats(hashable=hashable)
+
+    def get_fighter_stats(self):
```

Comment:
`get_fighter_stats` looks like pair for `get_basic_stats` but have different signature, it looks strange and unexpected.
 May be always return hashable value in both cases?